MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/analytics/AnalyticsTracker.java

```diff
@@ -0,0 +1,215 @@
+package com.mapbox.mapboxandroiddemo.analytics;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.os.Build;
+import android.preference.PreferenceManager;
+import android.support.annotation.NonNull;
+
+import com.mapbox.mapboxandroiddemo.R;
+import com.segment.analytics.Analytics;
+import com.segment.analytics.messages.IdentifyMessage;
+import com.segment.analytics.messages.ScreenMessage;
+import com.segment.analytics.messages.TrackMessage;
+
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+
+/**
+ * This class abstracts various analytics calls to Segment analytics' Java library.
+ */
+
+public class AnalyticsTracker {
+
+  private Context appContext = null;
+
+  private static volatile AnalyticsTracker analyticsInstance;
+  private static volatile Analytics analytics;
+  private static final String CLICKED_ON_NAV_DRAWER_SECTION_EVENT_NAME = "Clicked on nav drawer section";
+  private static final String CLICKED_ON_INDIVIDUAL_EXAMPLE_EVENT_NAME = "Clicked on individual example";
+  private static final String SECTION_NAME_MAP_KEY = "section name";
+  private static final String EXAMPLE_NAME_MAP_KEY = "example name";
+  private static final String IS_TABLET_MAP_VALUE = "tablet";
+  private static final String IS_PHONE_MAP_VALUE = "phone";
+  private static final String MAPBOX_SHARED_PREFERENCE_KEY_ANALYTICS_ENABLED = "mapboxAnalyticsEnabled";
+  private static final String MAPBOX_SHARED_PREFERENCES_FILE = "MapboxSharedPreferences";
+  private static String MAPBOX_USERNAME;
+  private Boolean analyticsEnabled = null;
+
+  /**
+   * Initializes instance of AnalyticsTracker class
+   **/
+  public static AnalyticsTracker getInstance(@NonNull Context context) {
+    if (analyticsInstance == null) {  // Single check
+      synchronized (AnalyticsTracker.class) {
+        if (analyticsInstance == null) {  // Double check
+          analyticsInstance = new AnalyticsTracker();
+          analyticsInstance.appContext = context;
+          analytics = Analytics.builder(context.getString(R.string.mapbox_segment_write_key)).build();
+        }
+      }
+    }
+    return analyticsInstance;
+  }
+
+  public void setMapboxUsername() {
+    MAPBOX_USERNAME = PreferenceManager.getDefaultSharedPreferences(
+        analyticsInstance.appContext).getString("USERNAME", "not logged in");
+  }
+
+  /**
+   * Gets and adds device information to analytics call. Ideally, this method is called
+   * when app is opened for the first time or if shared preferences is cleared.
+   **/
+
+  public void openedAppForFirstTime(boolean isTablet, boolean loggedIn) {
+    if (isAnalyticsEnabled()) {
+      Map<String, String> properties = new HashMap<>();
+      properties.put("email", getSharedPreferences(appContext)
+          .getString("EMAIL", "not logged in"));
```

Comment:
I should be using a constant string instead of `"EMAIL"` , no? If so, my `EMAIL_KEY` constant string is all caps, so there won't be consistency...